ui/components/app/approve-content-card/approve-content-card.js

```diff
@@ -0,0 +1,241 @@
+                    textAlign={TEXT_ALIGN.RIGHT}
+                  >
+                    <Box>
+                      <Typography
+                        variant={TYPOGRAPHY.H4}
+                        fontWeight={FONT_WEIGHT.BOLD}
+                        color={COLORS.TEXT_DEFAULT}
+                      >
+                        {formatCurrency(fiatTransactionTotal, currentCurrency)}
+                      </Typography>
+                    </Box>
+                    <Box>
+                      <Typography
+                        variant={TYPOGRAPHY.H6}
+                        fontWeight={FONT_WEIGHT.NORMAL}
+                        color={COLORS.TEXT_MUTED}
+                      >
+                        {`${ethTransactionTotal} ${nativeCurrency}`}
+                      </Typography>
+                    </Box>
+                  </Box>
+                </>
+              )}
+            </Box>
+          ))}
+        {renderDataContent && (
+          <Box display={DISPLAY.FLEX} flexDirection={FLEX_DIRECTION.COLUMN}>
+            <Box>
+              <Typography
+                variant={TYPOGRAPHY.H7}
+                color={COLORS.TEXT_ALTERNATIVE}
+              >
+                {isSetApproveForAll
+                  ? t('functionSetApprovalForAll')
+                  : t('functionApprove')}
+              </Typography>
+            </Box>
+            {isSetApproveForAll && isApprovalOrRejection !== undefined ? (
+              <Box>
+                <Typography
+                  variant={TYPOGRAPHY.H7}
+                  color={COLORS.TEXT_ALTERNATIVE}
+                >
+                  {`${t('parameters')}: ${isApprovalOrRejection}`}
+                </Typography>
+              </Box>
+            ) : null}
+            <Box
+              marginRight={4}
+              className="approve-content-card-container__data__data-block"
+            >
+              <Typography
+                variant={TYPOGRAPHY.H7}
+                color={COLORS.TEXT_ALTERNATIVE}
+              >
+                {data}
+              </Typography>
+            </Box>
+          </Box>
+        )}
+      </Box>
+      {footer}
+    </Box>
+  );
+}
+
+ApproveContentCard.propTypes = {
+  showHeader: PropTypes.bool,
+  symbol: PropTypes.node,
+  title: PropTypes.string,
+  showEdit: PropTypes.bool,
+  showAdvanceGasFeeOptions: PropTypes.bool,
+  onEditClick: PropTypes.func,
+  footer: PropTypes.node,
+  noBorder: PropTypes.bool,
+  supportsEIP1559V2: PropTypes.bool,
+  renderTransactionDetailsContent: PropTypes.bool,
+  renderDataContent: PropTypes.bool,
+  isMultiLayerFeeNetwork: PropTypes.bool,
+  ethTransactionTotal: PropTypes.string,
+  nativeCurrency: PropTypes.string,
+  fullTxData: PropTypes.object,
+  hexTransactionTotal: PropTypes.string,
+  fiatTransactionTotal: PropTypes.string,
+  currentCurrency: PropTypes.string,
+  isSetApproveForAll: PropTypes.bool,
+  isApprovalOrRejection: PropTypes.bool,
+  data: PropTypes.string,
```

Comment:
PropType descriptions would be nice here

Reply:
I'm gonna squash and merge, @filipsekulic please add those prop type descriptions in the same pr you mentioned above.

Reply:
Agreed :+1: